File: implementacao java/src/Dominio/Candidato.java
package Dominio;
import java.time.LocalDate;
import java.time.Period;


public class Candidato implements Comparable<Candidato>{

    private int codigoMunicipio;
    private int codigoCargo;
    private int numeroCandidato;
    private String nomeUrna;
    private int numeroFederacao;
    private LocalDate dataNascimento;
    private boolean resultadoTurno;
    private Genero genero;
    private int idade;
    private Partido partido;
    private int qtdVotosRecebidos;
    
    

    public Candidato(int codigoMunicipio, int codigoCargo, int numeroCandidato, String nomeUrna, int numeroFederacao, LocalDate dataNascimento, boolean resultadoTurno, Genero genero,  Partido partido, LocalDate dataAtual) {

        this.codigoMunicipio = codigoMunicipio;
        this.codigoCargo = codigoCargo;
        this.numeroCandidato = numeroCandidato;
        this.nomeUrna = nomeUrna;
        this.numeroFederacao = numeroFederacao;
        this.dataNascimento = dataNascimento;
        this.resultadoTurno = resultadoTurno;
        this.genero = genero;
        this.partido = partido;
        this.idade = calculaIdade(dataAtual);
       
    }
    
    public boolean getResultadoTurno() {
        return resultadoTurno;
    }
    public int getNumeroFederacao() {
        return numeroFederacao;
    }

    public Genero getGenero() {
        return genero;
    }

    public int getIdade() {
        return idade;
    }

    public int getQtdVotosRecebidos() {
        return qtdVotosRecebidos;
    }

    public LocalDate getDataNascimento() {
        return dataNascimento;
    }

    public String getNomeUrna() {
        return nomeUrna;
    }
    
    
    public int getNumeroCandidato() {
        return numeroCandidato;
    }
    
    
    public Partido getPartido() {
        return partido;
    }

    
    public int getQtdVotos(){
        return this.qtdVotosRecebidos;
    }

    
    public void incrementarVotos(int qtd){
        this.qtdVotosRecebidos += qtd;
        this.partido.atualizaAtributos(qtd, 0);
    }

    private int calculaIdade(LocalDate dataAtual)
    {   
     return Period.between(this.dataNascimento, dataAtual).getYears();
    }

    @Override
    public int compareTo(Candidato candidato){
        int compCriterioI = Integer.compare(candidato.getQtdVotos(), this.qtdVotosRecebidos);
        
        if(compCriterioI != 0){
            return compCriterioI;
        }

        return this.dataNascimento.compareTo(candidato.getDataNascimento());
    }
}

File: implementacao c++/Candidato.cpp
#include "Candidato.hpp"
#include "Partido.hpp"

Candidato::Candidato(int codigoMunicipio, int codigoCargo, int numeroCandidato, const std::string &nomeUrna, int numeroFederacao, SituacaoCandidato resultadoTurno, Genero &genero,  Partido *partido, Data& dataNascimento,  Data &dataAtual)
    : codigoMunicipio(codigoMunicipio),      
      codigoCargo(codigoCargo),             
      numeroCandidato(numeroCandidato),      
      nomeUrna(nomeUrna),
      numeroFederacao(numeroFederacao),
      dataNascimento(dataNascimento),     
      resultadoTurno(resultadoTurno),
      genero(genero),
      partido(partido),
      qtdVotosRecebidos(0)                 
{
    this->idade = calculaIdade(dataAtual);
}

const std::string& Candidato::getNome() const{
    return this->nomeUrna;
}

int Candidato::getVotos() const{
    return this->qtdVotosRecebidos;
}

int Candidato::getNumeroCandidato() const{
    return this->numeroCandidato;
}

int Candidato::getNumeroFederacao() const{
    return this->numeroFederacao;
}

Partido* Candidato::getPartido() const{
    return this->partido;
}

SituacaoCandidato Candidato::getResultadoTurno() const{
    return this->resultadoTurno;
}

void Candidato::incrementarVotos(int votos){

    this->qtdVotosRecebidos += votos;
    this->partido->atualizaAtributos(votos, 0);
}


bool Candidato::operator<(Candidato const &outro)const{

    if (this->getVotos() != outro.getVotos()) {

        return this->getVotos() > outro.getVotos();
    }

    return this->getIdade() > outro.getIdade();
 }

bool Candidato::operator==(Candidato const&outro) const{
    return this->getNumeroCandidato() == outro.getNumeroCandidato();
}

int Candidato::getIdade() const{
    return this->idade;
}

int Candidato::calculaIdade(const Data& dataAtual) const{
 
    auto time_point_nascimento = this->dataNascimento.getTimePoint();
    auto time_point_eleicao = dataAtual.getTimePoint();

    std::time_t t_nascimento = std::chrono::system_clock::to_time_t(time_point_nascimento);
    std::time_t t_eleicao = std::chrono::system_clock::to_time_t(time_point_eleicao);

    std::tm tm_nascimento = *std::localtime(&t_nascimento);
    std::tm tm_eleicao = *std::localtime(&t_eleicao);

    int anos = tm_eleicao.tm_year - tm_nascimento.tm_year;
    
    if (tm_eleicao.tm_mon < tm_nascimento.tm_mon || 
       (tm_eleicao.tm_mon == tm_nascimento.tm_mon && tm_eleicao.tm_mday < tm_nascimento.tm_mday)) {
        anos--; 
    }
    
    return anos;
}